Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbNativeWindow.cpp

```diff
@@ -219,17 +231,143 @@ namespace AzFramework
         }
     }
 
+
+    // This function will draw the splash image if one is set
+    // otherwise it will return without doing anything
+    void XcbNativeWindow::DrawSplash()
+    {
+#ifndef SPLASH_IMAGE_EXISTS
+    return;  // failed to find splash image no need to draw
+#else
+        const unsigned int bytesPerPixel = 4;
+        
+        typedef struct
+        {
+            const png_byte* data;
+            const png_size_t size;
+        } DataHandle;
+
+        typedef struct
+        {
+            const DataHandle data;
+            png_size_t offset;
+        } ReadDataHandle;
+
+        auto ReadDataFromInputStream = [](
+            png_structp pngPtr,
+            png_byte* rawData,
+            png_size_t readLength)
+        {
+            ReadDataHandle* handle = (ReadDataHandle*)png_get_io_ptr(pngPtr);
```

Comment:
use C++ style casting
```suggestion
            ReadDataHandle* handle = static_cast<ReadDataHandle*>(png_get_io_ptr(pngPtr));
```